content/articles/how-is-textInputLayout-different-from-edittext/index.md

```diff
@@ -0,0 +1,304 @@
+
+### Text field
+A text field is a standard entry widget (single line) that when a user touches it, it places a cursor on it and displays a keyboard hence allowing user to type text into your app.  
+To add a text field to your layout we use EditText object.
+
+### Using text field
+To use text field in your android application.
+First, Import or add the new material components dependency.
+
+### Filled text field
+According to official documentation, filled text field have more visual emphasis than outlined text field and hence making them stand out when surrounded by other content and components.
+It has two height variants, that is it consist of standard and dense text fields.
+Filled text field is the default styles if the style is not set.
+
+### Outlined text field
+ In outlined text filled we apply styles on the TextInputLayout. As a result we will  get the outlined look text field.
+Like filled text field, outlined text field has two height variants. that is, it consists of standard and dense text fields.
+We have attributes that you can use to set the outlined look (like corner radius outline and stroke color outline) of a text field.
+
+### Theming text field
+In theming text field, we use material theming in text fields. You can be customize  text field in term of color, topography and shape.
+Like to filled text field and outlined text filled, theming text field has two height variants, that is, it consists of standard and dense text fields.
+
+### Adding TextInputLayout to our android app.
+Now, I am going to create a new project to show you how to work with TextInputLayout in an android application.I will be using a single activity application and we will do  all cool stuff inside activity_main.
+
+### Step 1:
+Creating Project
+First you have to open a new project in your android studio. Then open your build.gradle . Add material dependency and click sync now.
+
+### Step 2: Adding material dependency
+```gradle 
+//Material  dependency 
+implementation 'com.google.android.material:material:1.4.0'
+``` 
+
+### Step 3: Adding TextInputLayout to your activity_main file
+Then we reach to the activity_main file and delete the default TextView add TextInputLayout as a child layout inside ConstraintLayout and constraint it and give it some margin.
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+    <com.google.android.material.textfield.TextInputLayout
+        android:layout_width="409dp"
+        android:layout_height="wrap_content"
```

Comment:
73: Using hard-coded dimensions will result in an unresponsive UI. Consider using `match_parent` for the width.